### ISSUE TITLE
Stage manager behaviour totally broken

### ISSUE THREAD

**Issue by murdock-grewar** [rxhanson/Rectangle#1063]
- [x] This is a new bug that hasn't been documented in a closed issue or in the Readme.

macOS version: Ventura 13.2
Rectangle version: 0.65

Hi, I love this app in principle, but unfortunately the Rectangle app just isn't resizing things to the correct window width with Stage Manager on. I've assigned hotkeys for snapping to the left 1/2 and right 1/2. But nothing ever seems to resize correctly. I know this is vague, but the whole functionality just seems broken in general.

The behaviour is inconsistent, but it fails more often that it works. Rectangle will seemingly randomly decide whether it should leave that space for the recent apps on the right hand side. This is particularly annoying because it means that the left and right halves often overlap, because one of them accounts for Stage Manager and the other does not.

Similar issues plague the other resizing options too, like centreing a window or almost-maximising it, even just maximising it.

The way I would describe the behaviour is that there seems to be a small chance (say, 20%) that Rectangle correctly leaves room for Stage Manager on the left-hand side. The rest of the time, it fails to leave room.

I love this app otherwise, but this bug makes it really frustrating to use. I can't see any pattern in the behaviour; it seems like a random chance.

**Comment by rxhanson** [comment]
Thanks for reporting. In order to figure out what's going on, it might take some additional logging beyond what is currently in the app. Would you be up for testing out another build with additional logging?

**Comment by murdock-grewar** [comment]
> Thanks for reporting. In order to figure out what's going on, it might take some additional logging beyond what is currently in the app. Would you be up for testing out another build with additional logging?

Yes certainly, that is easy for me. If you can tell me how to launch Rectangle via the terminal so that it prints to `stdout` and `stderr`, that would be preferable. I can pipe those to log files and send them to you.

**Comment by murdock-grewar** [comment]
Additional note: I have disconnected from the external monitor, and currently Rectangle seems to be working well. The issue might only occur when plugged into another monitor. However, the issue then affects both monitors.

**Comment by rxhanson** [comment]
Thanks! This actually points to a macOS issue, then. Rectangle doesn't handle displays uniquely in a way that could cause this. There's also this issue that was recently added: https://github.com/rxhanson/Rectangle/issues/1062. Without digging deep into it, my first guess is that Ventura has a bug where external displays may affect the Accessibility API that Rectangle relies on for obtaining and setting window positions and sizes.

**Comment by lukasmaxim** [comment]
First of all, thanks a lot for this window manager. Was on the lookout for one that supports having stage manager for a while and Rectangle is just what I needed!

Just adding here that the issue still exists on

macOS version: Ventura 13.4
Rectangle version: 0.69

Behavior is consistent with what murdock-grewar describes, shortcuts for maximizing, 1/2 splitting work fine with stage manager on on a single display, but are inconsistent (more often than not won't work) when having a second monitor plugged in. Would be really great if there was a way to investigate this. I'm happy to send additional info if that helps!

**Comment by jpatrolla** [comment]
Hi everyone. Had the same problem. Looks to be a conflict with a macOS setting.

System Settings > Desktop & Dock > Double-click a window's title bar to **Do Nothing**

Seems to be fixed so far for me. Hope that helps

**Comment by jpatrolla** [comment]
Hi gang. Try changing the following setting to 'Do Nothing':

![image](https://github.com/rxhanson/Rectangle/assets/9730683/987b3465-a620-4f47-b2d7-3820c5730148)

System Settings > Desktop & Dock > Double-click a window's title bar to **Do Nothing**

Seems to be fixed so far for me. Hope that helps

**Comment by pereljon** [comment]
Dragging to the left incorporates the window into Stage Manager even though the Rectangle "preview" window outline is visible. Would expect that dragging to the left would resize the window in the current Stage Manager space.

**Comment by rxhanson** [comment]
Closing this one out since it's poised to be a catch-all for Stage Manager related issues. To recap my investigation: the root issue here appears to be a problem with how the accessibility API reports Stage Manager, particularly when there are multiple displays. Unfortunately, this pushes it to be something best fixed by Apple. As such, I don't have immediate plans to investigate further, but am open to anyone else taking a look.
